### ISSUE TITLE
Error: Failed to install provider :Error while installing eppo/environment v1.3.5: archive has incorrect checksum

### ISSUE THREAD

**Issue by skumar-16** [EppO/terraform-provider-environment#451]
We are not able to initialize  wncountering below error.

We tried to:-
1. delete the lock file
2. replace the expected checksum
3. tried to reinstall terraform
4. tried terraform providers lock  -platform=darwin_amd64
5. tried terraform providers lock
6. tried in windows os, same error occuring.

Still no help, could you please help us on it.

**1.** 
templates/config/main.tf

required_providers {
environment = {
source = "EppO/environment"
}
}

Error: Failed to install provider
│
│ Error while installing eppo/environment v1.3.5: archive has incorrect checksum
│ zh:26b31222d54be16500a184f8a946a357e5502f813c727bf6f2c7af360820a0ea (expected
│ zh:95fb102fecceb3a5b44dbe9fbe262494a0abdb6805addf1286c5d92cd4b0f779)

**2. we have some more log with debug.**

Error:
(base) skumar16@C02G67TSML7H aws % terraform init
2024-02-12T13:33:24.895-0800 [INFO] Terraform version: 1.5.7
2024-02-12T13:33:24.896-0800 [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2024-02-12T13:33:24.896-0800 [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2024-02-12T13:33:24.896-0800 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2024-02-12T13:33:24.896-0800 [DEBUG] using github.com/zclconf/go-cty v1.12.2
2024-02-12T13:33:24.896-0800 [INFO] Go runtime version: go1.21.1
2024-02-12T13:33:24.896-0800 [INFO] CLI args: []string{"terraform", "init"}
2024-02-12T13:33:24.897-0800 [DEBUG] Attempting to open CLI config file: /Users/skumar16/.terraformrc
2024-02-12T13:33:24.897-0800 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-02-12T13:33:24.897-0800 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-02-12T13:33:24.897-0800 [DEBUG] ignoring non-existing provider search directory /Users/skumar16/.terraform.d/plugins
2024-02-12T13:33:24.897-0800 [DEBUG] ignoring non-existing provider search directory /Users/skumar16/Library/Application Support/io.terraform/plugins
2024-02-12T13:33:24.897-0800 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2024-02-12T13:33:24.900-0800 [INFO] CLI command args: []string{"init"}

Initializing the backend...
2024-02-12T13:33:24.919-0800 [DEBUG] New state was assigned lineage "5a0104cc-1f84-8e1b-eee7-e4e9522ebf1e"
2024-02-12T13:33:25.405-0800 [DEBUG] checking for provisioner in "."
2024-02-12T13:33:25.490-0800 [DEBUG] checking for provisioner in "/usr/local/bin"
Initializing modules...
2024-02-12T13:33:25.494-0800 [DEBUG] Module installer: begin config
2024-02-12T13:33:25.500-0800 [DEBUG] Module installer: begin service

Initializing provider plugins...

Reusing previous version of hashicorp/aws from the dependency lock file
2024-02-12T13:33:25.528-0800 [DEBUG] Service discovery for registry.terraform.io at https://registry.terraform.io/.well-known/terraform.json
2024-02-12T13:33:25.776-0800 [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/aws/versions
Reusing previous version of eppo/environment from the dependency lock file
2024-02-12T13:33:25.949-0800 [DEBUG] GET https://registry.terraform.io/v1/providers/eppo/environment/versions
Using previously-installed hashicorp/aws v5.21.0
2024-02-12T13:33:26.497-0800 [DEBUG] GET https://registry.terraform.io/v1/providers/eppo/environment/1.3.5/download/darwin/amd64
2024-02-12T13:33:26.675-0800 [DEBUG] GET https://github.com/EppO/terraform-provider-environment/releases/download/v1.3.5/terraform-provider-environment_1.3.5_SHA256SUMS
2024-02-12T13:33:27.409-0800 [DEBUG] GET https://github.com/EppO/terraform-provider-environment/releases/download/v1.3.5/terraform-provider-environment_1.3.5_SHA256SUMS.sig
Installing eppo/environment v1.3.5...
╷
│ Error: Failed to install provider
│
│ Error while installing eppo/environment v1.3.5: archive has incorrect checksum
│ zh:26b31222d54be16500a184f8a946a357e5502f813c727bf6f2c7af360820a0ea (expected
│ zh:95fb102fecceb3a5b44dbe9fbe262494a0abdb6805addf1286c5d92cd4b0f779)
╵

mac OS:

`(base) skumar16@C02G67TSML7H aws % terraform init

Initializing the backend...
Initializing modules...

Initializing provider plugins...

Finding latest version of eppo/environment...
Finding latest version of hashicorp/aws...
Installing eppo/environment v1.3.5...
Installing hashicorp/aws v5.36.0...
Installed hashicorp/aws v5.36.0 (signed by HashiCorp)
╷
│ Error: Failed to install provider
│
│ Error while installing eppo/environment v1.3.5: archive has incorrect checksum
│ zh:26b31222d54be16500a184f8a946a357e5502f813c727bf6f2c7af360820a0ea (expected
│ zh:95fb102fecceb3a5b44dbe9fbe262494a0abdb6805addf1286c5d92cd4b0f779)
╵
(base) skumar16@C02G67TSML7H aws % pwd
/Users/skumar16/Library/CloudStorage/OneDrive-Illumina,Inc/Desktop/Artifactory/github-action/action-demo-terraform/git/git-action-elastic-runners/templates/aws
(base) skumar16@C02G67TSML7H aws % terraform providers lock

Fetching eppo/environment 1.3.5 for darwin_amd64...
Fetching hashicorp/aws 5.36.0 for darwin_amd64...
Retrieved hashicorp/aws 5.36.0 for darwin_amd64 (signed by HashiCorp)
╷
│ Error: Could not retrieve providers for locking
│
│ Terraform failed to fetch the requested providers for darwin_amd64 in order to calculate their checksums: some providers could not be
│ installed:
│ - registry.terraform.io/eppo/environment: archive has incorrect checksum
│ zh:26b31222d54be16500a184f8a946a357e5502f813c727bf6f2c7af360820a0ea (expected
│ zh:95fb102fecceb3a5b44dbe9fbe262494a0abdb6805addf1286c5d92cd4b0f779).
╵
(base) skumar16@C02G67TSML7H aws % `

windows OS

`terraform init

Initializing the backend...
Initializing modules...

config in ..\config
service in service
Initializing provider plugins...

Reusing previous version of eppo/environment from the dependency lock file
Reusing previous version of hashicorp/aws from the dependency lock file
Installing eppo/environment v1.3.5...
Installing hashicorp/aws v5.21.0...
Installed hashicorp/aws v5.21.0 (signed by HashiCorp)
╷
│ Error: Failed to install provider
│
│ Error while installing eppo/environment v1.3.5: archive has incorrect checksum zh:e4bf3383cb0b5b85eef10ac594a5f6f70f5b45abe7a1513b759181f58dda2dce (expected
│ zh:018d92e621177d053ed5c32e8220efa8c019852c4d60cc7539683bac28470d9b)
╵
PS C:\Users\sbhavsar\OneDrive - Illumina, Inc\Desktop\scalable-action-runner\git-action-elastic-runners\templates\aws> terraform provider locks
Terraform has no command named "provider". Did you mean "providers"?

To see all of Terraform's top-level commands, run:
terraform -help

PS C:\Users\sbhavsar\OneDrive - Illumina, Inc\Desktop\scalable-action-runner\git-action-elastic-runners\templates\aws> terraform providers locks
Too many command line arguments. Did you mean to use -chdir?
PS C:\Users\sbhavsar\OneDrive - Illumina, Inc\Desktop\scalable-action-runner\git-action-elastic-runners\templates\aws> terraform providers lock

Fetching hashicorp/aws 5.21.0 for windows_386...
Retrieved hashicorp/aws 5.21.0 for windows_386 (signed by HashiCorp)
Fetching eppo/environment 1.3.5 for windows_386...
╷
│ Error: Could not retrieve providers for locking
│
│ Terraform failed to fetch the requested providers for windows_386 in order to calculate their checksums: some providers could not be installed:
│ - registry.terraform.io/eppo/environment: archive has incorrect checksum zh:e4bf3383cb0b5b85eef10ac594a5f6f70f5b45abe7a1513b759181f58dda2dce (expected zh:018d92e621177d053ed5c32e8220efa8c019852c4d60cc7539683bac28470d9b).
╵

**Comment by EppO** [comment]
I just released a new version 1.3.6 and tried a terraform init/plan on darwin amd64, it went ok. Here is my lineup:
```
Terraform v1.8.1
on darwin_amd64
+ provider registry.terraform.io/e-breuninger/netbox v3.8.5
+ provider registry.terraform.io/eppo/environment v1.3.6
+ provider registry.terraform.io/hashicorp/azuread v2.48.0
+ provider registry.terraform.io/hashicorp/azurerm v3.100.0
+ provider registry.terraform.io/hashicorp/http v3.4.2
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

Can you please try again with 1.3.6?

**Comment by EppO** [comment]
please re-open if the issue is persists with latest versions
